pkg/apis/providerconfig/v1alpha1/types.go

```diff
@@ -25,66 +25,22 @@ import (
 
 // ExternalMachineProviderConfig provides machine configuration struct
 type ExternalMachineProviderConfig struct {
-	metav1.TypeMeta `json:",inline"`
+	metav1.TypeMeta   `json:",inline"`
 
-	// A list of roles for this Machine to use.
```

Comment:
Why is all this stuff getting deleted?

Reply:
You're forgetting that we will eventually need a metal provider that does actual provisioning, which would make OS a useful field. Its not hard to imagine physical machines being grouped in some way, perhaps into "Zones".
The Roles field is also important in that the OCP folks have already identified that they never ever want to fence a master.
These fields are also how we look for the correct machine template to use, which you also deleted.

Please undo and do not make the mistake of assuming everything is a blind copy of the GCE repo.
Some elements may not be in their final form - the actuator is far from finished, but there were reasons for keeping the parts that remain.

Reply:
If you gonna to re-provision bare metal hosts via Foreman(and currently I do not see other option), you will need more fields related to Foreman host entity, but again I prefer to add it when we will have such requirement, I can not predict the future so I prefer to work with today requirements, we do not have complex architecture under actuator and we also only introduce API version alpha that can be changed a lot, but better to start from small set of fields and not overload our API with fields that we do not use under the code.
You can ask someone from the side about API, and you will see that a lot of fields very confusing.